symfony/client/src/orangehrmAdminPlugin/pages/qualificationSkill/EditQualificationSkill.vue

```diff
@@ -0,0 +1,168 @@
+<!--
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+ -->
+
+<template>
+  <div class="orangehrm-background-container">
+    <div class="orangehrm-card-container">
+      <oxd-text tag="h6">Edit Skill</oxd-text>
+
+      <oxd-divider />
+
+      <oxd-form :loading="isLoading" @submitValid="onSave">
+        <oxd-form-row>
+          <oxd-input-field
+            label="Skill Name"
+            v-model="skill.name"
+            :rules="rules.name"
+            required
+          />
+        </oxd-form-row>
+
+        <oxd-form-row>
+          <oxd-input-field
+            type="textarea"
+            label="Skill Description"
+            placeholder="Type description here"
+            v-model="skill.description"
+            :rules="rules.description"
+          />
+        </oxd-form-row>
+
+        <oxd-divider />
+
+        <oxd-form-actions>
+          <oxd-button displayType="ghost" label="Cancel" @click="onCancel" />
+          <oxd-button
+            class="orangehrm-left-space"
+            displayType="secondary"
+            label="Save"
+            type="submit"
+          />
+        </oxd-form-actions>
+      </oxd-form>
+    </div>
+  </div>
+</template>
+
+<script>
+import {navigate} from '@orangehrm/core/util/helper/navigation';
+import {APIService} from '@/core/util/services/api.service';
+
+const skillModel = {
+  id: '',
+  name: '',
+  description: '',
+};
+
+export default {
+  props: {
+    qualificationSkillId: {
+      type: Number,
+      required: true,
+    },
+  },
+
+  data() {
+    return {
+      isLoading: false,
+      skill: {...skillModel},
+      rules: {
+        name: [],
+        description: [
+          v =>
+            (v && v.length < 400) ||
```

Comment:
maybe we could move into generic function and import in this component `validCharLength(400)`